packages/frontend/src/app/modules/common/components/form-controls/interpolating-text-form-control/interpolating-text-form-control.component.ts

```diff
@@ -0,0 +1,268 @@
+import { BooleanInput, coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+	ChangeDetectionStrategy,
+	Component,
+	DoCheck,
+	HostBinding,
+	Input,
+	OnDestroy,
+	OnInit,
+	Optional,
+	Self,
+	ViewChild,
+} from '@angular/core';
+import { ControlValueAccessor, FormControl, FormGroupDirective, NgControl, NgForm, Validators } from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+import { MatFormFieldControl } from '@angular/material/form-field';
+import { QuillEditorComponent, QuillModules } from 'ngx-quill';
+import { lastValueFrom, Observable, skip, Subject, take, tap } from 'rxjs';
+import {
+	CustomErrorMatcher,
+	fromOpsToText,
+	fromTextToOps,
+	InsertMentionOperation,
+	QuillEditorOperationsObject,
+	QuillMaterialBase,
+	TextInsertOperation,
+} from './utils';
+import 'quill-mention';
+import { Store } from '@ngrx/store';
+import { BuilderSelectors } from 'src/app/modules/flow-builder/store/builder/builder.selector';
+
+@Component({
+	selector: 'app-interpolating-text-form-control',
+	templateUrl: './interpolating-text-form-control.component.html',
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: MatFormFieldControl,
+			useExisting: InterpolatingTextFormControlComponent,
+		},
+		{
+			provide: ErrorStateMatcher,
+			useClass: CustomErrorMatcher,
+		},
+	],
+})
+export class InterpolatingTextFormControlComponent
```

Comment:
Is it possible to not add space after the variable, that will cause an issues when you want to input single variable.

Single variable get resolved to their orignal type, adding extra spaces or text will cast it to string.